modules/gateway/peers_test.go

```diff
@@ -653,6 +653,8 @@ func TestDisconnect(t *testing.T) {
 		if err != nil {
 			t.Fatal("accept failed:", err)
```

Comment:
Not changed in this PR, but `t.Fatal` and `t.Error` cannot be called from a goroutine. There are probably many other places we do this, but my regex-foo is not strong enough to find them easily.